### PR TITLE
Type Casting in Java

### LINKED ISSUE #2075: Type Casting in Java
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Type Casting in Java

### Introduction paragraph (2-3 paragraphs):
Type casting is a process of changing one type value to another type. It is also known as type conversion or Type Coercion.
There are two types of casting in java, that is explicit and implicit casting. 
Explicit casting is when you are assigning a larger type value to a variable of smaller type while Implicit casting is when you are assigning a smaller type value to a variable of larger type. In this article, we are going to cover the two type of casting and simple programs on how to go about it.

### Key takeaways: 

1. What is type casting
2. Explicit casting
3. Implicit casting
4. Advantages
5. Disadvantages

### References: 
N/A

### PR DESCRIPTION
**Pull request by Terripha**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes #2075

### CODE REVIEW

**Comment by zolomohan** [comment]
Hi @Terripha. Good Job with the article. The article passed the preliminary review. Moving on to tech review.

File: content/articles/type-casting-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+In the example above, we have converted the double data type into an `int` data type. The decimal part of the value is lost after type casting.
+
+### Java wrapper class
+
+To convert a primitive data type into an object, we need the [wrapper classes](https://beginnersbook.com/2017/09/wrapper-class-in-java/). We can take `string` inputs from the users e.g "1234".  If we want to work with integers, we need to convert the `string` to `int` data type and vice versa. i.e
+
+```Java
+string a = "1234";
```
Comment (zolomohan) [inline content/articles/type-casting-in-java/index.md:97]:
```suggestion
String a = "1234";
```

File: content/articles/type-casting-in-java/index.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+**Example**
+
+```java
+public class Main {
+  public static void main (String args[]) {
+      
+    int num = 67;
+
+    Double mydouble = new Double(num); // using Double wrapper class
```
Comment (zolomohan) [inline content/articles/type-casting-in-java/index.md:120]:
```suggestion
    Double myDouble = new Double(num); // using Double wrapper class
```

**Review by zolomohan — CHANGES_REQUESTED** [review]
Hi @Terripha. Please go through and fix these minor errors in the code.

**Comment by Terripha** [comment]
@zolomohan  i have updated the changes

**Review by zolomohan — APPROVED** [review]

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by ahmadmardeni1** [comment]
Great article @Terripha 👍
Congrats on your first published article! ✨

Please be sure to go through the minor edits I made.
@zolomohan please do the same  🚀

**Pull request merged**